server/src/shared/infra/cache/redis-repository.ts

```diff
@@ -0,0 +1,64 @@
+import { RedisClient } from "../../../setup/cache/redis"
+import { AppError } from "../../core/app-error"
+import { Result } from "../../core/result"
+import { RedisEntity } from "./redis-entity"
+
+export type RedisSaveEntityResponse = Result<RedisSaveEntitySuccess, RedisSaveEntityError>
+export type RedisSaveEntitySuccess = boolean
+export type RedisSaveEntityError = AppError.UnexpectedError
+
+export type RedisGetEntityResponse<RedisEntityType> = Result<RedisEntityType, RedisGetEntityError>
+export type RedisGetEntityError = AppError.UnexpectedError
+
+export type RedisDeleteEntityResponse = Result<RedisDeleteEntitySuccess, RedisDeleteEntityError>
+export type RedisDeleteEntitySuccess = boolean
+export type RedisDeleteEntityError = AppError.UnexpectedError
+export interface RedisSetOptions {
+    mode: string
+    value: number
+}
+
+export class RedisRepository<RedisEntityType> {
+
+    async getEntity(entityKey: string): Promise<RedisGetEntityResponse<RedisEntityType>>{
+        return new Promise((resolve) => {
+            RedisClient().get(entityKey, (err, value) => {
+                if(err || value === null){
+                    resolve(Result.err(new AppError.UnexpectedError("Redis get operation failed.")))
```

Comment:
There are quite a few "if err then resolve Result.err('something bad')" patterns in this file. Can we recover any kind of message from the Redis errors? If we don't return it in the AppError, we should at least log it.